RemoveResets doesn't handle uninitialized non-self inits

`RemoveResets` doesn't properly remove resets connected to ground that are not-initialized to themselves. `RemoveResets` will then *not* remove these resets and the Verilog emitter will error on them.

[`RemoveResets` treats a reset connection to ground as sufficient to say that the register requires no further processing.](https://github.com/freechipsproject/firrtl/blob/c9d49aa8a600a59f3cfc1dcfc9ec8729077d5107/src/main/scala/firrtl/transforms/RemoveReset.scala#L60)

You can reproduce this with the following circuit:

```
circuit Foo:
  module Foo:
    input clock: Clock

    input a: UInt<1>
    output b: UInt<1>

    reg r: UInt<1>, clock with:
      reset => (UInt<1>(0), UInt<1>(1))
    r <= a
    b <= r
```

```
Exception in thread "main" java.lang.AssertionError: assertion failed: Error! Synchronous reset should have been removed!
	at scala.Predef$.assert(Predef.scala:223)
	at firrtl.VerilogEmitter$VerilogRender.regUpdate(Emitter.scala:665)
	at firrtl.VerilogEmitter$VerilogRender.build_streams(Emitter.scala:826)
	at firrtl.VerilogEmitter$VerilogRender.$anonfun$build_streams$1(Emitter.scala:812)
	at firrtl.VerilogEmitter$VerilogRender.$anonfun$build_streams$1$adapted(Emitter.scala:812)
```

### Checklist

- [x] Did you specify the current behavior?
- [x] Did you specify the expected behavior?
- [x] Did you provide a code example showing the problem?
- [x] Did you describe your environment?
- [x] Did you specify relevant external information?

### What is the current behavior?

Valid FIRRTL circuit errors in the Verilog emitter.

### What is the expected behavior?

One of:
1. This should be an error in an early `Check`
2. The init should be removed in `RemoveResets`

This should not be an error in the Verilog emitter.

### Steps to Reproduce

<!-- How can someone else reproduce the problem you're seeing? -->
<!-- It's very helpful to include a full example of a failing Chisel or FIRRTL program! -->
<!-- Include a stack trace if you have it! -->

Put the FIRRTL example above in a file and run:

```
firrtl -i Foo.fir
```

### Your environment

FIRRTL 1.3.0

### External Information

<!-- Was this discussed anywhere else, e.g., Twitter, Gitter, StackOverflow? Provide direct links if available! -->

None.

--- a/src/main/scala/firrtl/transforms/RemoveReset.scala
+++ b/src/main/scala/firrtl/transforms/RemoveReset.scala
@@ -56,8 +56,9 @@ object RemoveReset extends Transform with DependencyAPIMigration {
         /* A register is initialized to an invalid expression */
         case reg @ DefRegister(_, _, _, _, _, init) if invalids.contains(we(init)) =>
           reg.copy(reset = Utils.zero, init = WRef(reg))
-        case reg @ DefRegister(_, rname, _, _, reset, init)
-            if reset != Utils.zero && reset.tpe != AsyncResetType =>
+        case reg @ DefRegister(_, rname, _, _, Utils.zero, _) =>
+          reg.copy(init = WRef(reg)) // canonicalize
+        case reg @ DefRegister(_, rname, _, _, reset, init) if reset.tpe != AsyncResetType =>
           // Add register reset to map
           resets(rname) = Reset(reset, init)
           reg.copy(reset = Utils.zero, init = WRef(reg))

--- a/src/test/scala/firrtlTests/transforms/RemoveResetSpec.scala
+++ b/src/test/scala/firrtlTests/transforms/RemoveResetSpec.scala
@@ -8,7 +8,7 @@ import firrtl.testutils.FirrtlFlatSpec
 import firrtl.testutils.FirrtlCheckers._
 
 import firrtl.{CircuitState, WRef}
-import firrtl.ir.{Connect, Mux}
+import firrtl.ir.{Connect, Mux, DefRegister}
 import firrtl.stage.{FirrtlCircuitAnnotation, FirrtlSourceAnnotation, FirrtlStage}
 
 class RemoveResetSpec extends FirrtlFlatSpec with GivenWhenThen {
@@ -112,4 +112,24 @@ class RemoveResetSpec extends FirrtlFlatSpec with GivenWhenThen {
     outputState shouldNot containTree { case Connect(_, WRef("foo_b",_,_,_), Mux(_,_,_,_)) => true }
   }
 
+  it should "canvert a reset wired to UInt<0> to a canonical non-reset" in {
+    Given("foo's reset is connected to zero")
+    val input =
+      """|circuit Example :
+         |  module Example :
+         |    input clock : Clock
+         |    input rst : UInt<1>
+         |    input in : UInt<2>
+         |    output out : UInt<2>
+         |    reg foo : UInt<2>, clock with : (reset => (UInt(0), UInt(3)))
+         |    foo <= in
+         |    out <= foo""".stripMargin
+
+    val outputState = toLowFirrtl(input)
+
+    Then("foo has a canonical non-reset declaration after RemoveReset")
+    outputState should containTree { case DefRegister(_, "foo", _,_, firrtl.Utils.zero, WRef("foo", _,_,_)) => true }
+    And("foo is NOT connected to a reset mux")
+    outputState shouldNot containTree { case Connect(_, WRef("foo",_,_,_), Mux(_,_,_,_)) => true }
+  }
 }

